docs/blueprints/EP036.rst

```diff
@@ -0,0 +1,226 @@
+:EP: 36
+:Title: Openflow multi-table pipeline processing
+:Authors:
+    - Aldo Ortega <aortegay AT fiu DOT edu>
+    - Vinicius Arcanjo <vindasil AT fiu DOT edu>
+    - Italo Valcy <idasilva AT fiu DOT edu>
+
+:Created: 2022-05-04
+:Updated: 2022-06-04
+:Kytos-Version: 2022.3
+:Status: Draft
+
+***************************************************
+EP36 - Openflow multi-table pipeline processing
+***************************************************
+
+Abstract
+========
+
+This blueprint will present the requirements for the implementation of switch multi-tables through new NApp called **multi_table**.
+
+Motivation
+==========
+
+Multi-table processing is desireable to take advantage of a pipelined processing of packets within a switch. It is necessary to ensure the consistency of tables to avoid missing packets thus a NApp will take care of the table system by being aware of all the flows created from NApps and users.
+
+Specification
+=============
+
+Define the table to be used by NApps and to be used by users which are going to be configurable.
+Also, a new NApp **multi_table** that will initially have the following responsibilities:
+
+  - Know which table is enabled.
+  - Ensuring the existence of a reasonable table-miss flow
+  - Publish events to handle the assigned table of each NApp
+  - In charge of modifying and analyzing **pipelines** collection
+  - The cookie prefix is **0xAD** for its posted flows.
+
+Pipeline Status: Enabled, Disabled
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The status of a pipeline dictates wheter it is applied as the current multi table system in every switch. Once a pipeline is ``enabled``, the previous one will be automatically ``disabled`` ensuring that there is always 1 or 0 (default pipeline is not counted) pipelines enabled.

```

Comment:
We might also need a `error` state, for a pipeline config that has might have wrong/invalid table groups. Can you double check this case, and see if makes sense?

Unless we never allowed this to happen during a pipeline creation request by importing the settings table groups of each NApp, but this would be too intrusive in terms of NApps boundaries/responsibilities.

Reply:
I did not add the `error` status because a stored pipeline was always a pipeline that followed the rules of no cycles and table number always goes up. Another case would be when a pipeline specifies the existence a NApp like `coloring` but this is not installed/enabled. In this case, the absence of an answer event would have been ignored.
Maybe the `error` is added for a pipeline that was being enabled but for some reason it was not possible. What other situation we would mark a pipeline with an error status?

Reply:
Right. Yes, the only one that I can see right now would when a pipeline with an nonexistent table group, or a NApp that doesn't exist where to be created. However, thinking about this, if we structure the NApps in a way where they always upfront publish which table groups they support, then during a request `multi_table` will already have that information that way we wouldn't need to model `error` in this iteration. 

The other ones would be actual bad event content or wrong shape or data when trying to set a table group `kytos/<napp_name>.enable_table`, but that would be an implementation error, this last one here we have full control and is very unlikely to happen. So we could defer its implementation too. 

Do you see any other error cases? Should we go without them and introduce this new event to publish upfront which table group each NApp support?

Reply:
Nope, I do not see other case. With the new event there is no use for `error` status, for now.